agent/engine/task_manager.go

```diff
@@ -226,25 +226,32 @@ func (mtask *managedTask) emitCurrentStatus() {
 // the task. This involves waiting for previous stops to complete so the
 // resources become free.
 func (mtask *managedTask) waitForHostResources() {
-	if mtask.StartSequenceNumber != 0 && !mtask.GetDesiredStatus().Terminal() {
-		seelog.Infof("Managed task [%s]: waiting for any previous stops to complete. Sequence number: %d",
-			mtask.Arn, mtask.StartSequenceNumber)
-		othersStopped := make(chan bool, 1)
-		go func() {
-			mtask.taskStopWG.Wait(mtask.StartSequenceNumber)
-			othersStopped <- true
-		}()
-		for !mtask.waitEvent(othersStopped) {
-			if mtask.GetDesiredStatus().Terminal() {
-				// If we end up here, that means we received a start then stop for this
-				// task before a task that was expected to stop before it could
-				// actually stop
-				break
-			}
+	if mtask.StartSequenceNumber == 0 {
+		// This is the first transition on this host. No need to wait
+		return
+	}
+	if mtask.GetDesiredStatus().Terminal() {
+		// Task's desired status is STOPPED. No need to wait in this case either
+		return
+	}
+
+	seelog.Infof("Managed task [%s]: waiting for any previous stops to complete. Sequence number: %d",
+		mtask.Arn, mtask.StartSequenceNumber)
+	othersStopped := make(chan bool, 1)
+	go func() {
+		mtask.taskStopWG.Wait(mtask.StartSequenceNumber)
+		othersStopped <- true
+	}()
+	for !mtask.waitEvent(othersStopped) {
+		if mtask.GetDesiredStatus().Terminal() {
+			// If we end up here, that means we received a start then stop for this
+			// task before a task that was expected to stop before it could
```

Comment:
:) it's a right of initiation for anyone who's relatively new to this code base to read this and wonder what in the world is the meaning of this sentence. Since I've already merged this PR, we can pick this next time we touch this file

Reply:
Maybe this will help anyone who stumbles here: We want to process stop events (a task changes to `desired=STOPPED`) prior to start events (`desired=RUNNING`) so that any resources that are consumed by the `desired=STOPPED` task can be freed for the `desired=RUNNING` task.  This logic helps to enforce the ordering.